tests/lac/linear_operator_01b.cc

```diff
@@ -0,0 +1,139 @@
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+// Test for composite operations
+
+#include "../tests.h"
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+
+#define PRINTME(name, var) \
+  deallog \
+      << "Solution vector: " << name << ": " \
+      << var;
+
+using namespace dealii;
+
+int main()
+{
+  initlog();
+  deallog << std::setprecision(10);
+
+  // SparseMatrix:
+  {
+    const unsigned int rc=2;
+    SparsityPattern sparsity_pattern (rc, rc, 0);
+    sparsity_pattern.compress();
+
+    SparseMatrix<double> A (sparsity_pattern);
+    Vector<double> b (rc);
+    for (unsigned int i=0; i < rc; ++i)
+      {
+        A.diag_element(i) = 5.0;
+        b(i) = 1.0;
+      }
+
+    const auto lo_A = linear_operator(A);
+    const auto lo_A_t = transpose_operator(lo_A);
+
+    SolverControl solver_control_A (100, 1.0e-10);
+    SolverCG< Vector<double> > solver_A (solver_control_A);
+    PreconditionJacobi< SparseMatrix<double> > preconditioner_A;
+    preconditioner_A.initialize(A);
+    const auto lo_A_inv = inverse_operator(lo_A,
+                                           solver_A,
+                                           preconditioner_A);
+    const auto lo_A_inv_t = transpose_operator(lo_A_inv);
+
+    deallog << "Normal and inverse multiplication operations" << std::endl;
+
+    const Vector<double> x1 = lo_A*b;
+    const Vector<double> x2 = lo_A_t *b;
+    const Vector<double> x3 = lo_A_inv*b;
+    const Vector<double> x4 = lo_A_inv_t *b;
+
+    PRINTME("x1", x1);
+    PRINTME("x2", x2);
+    PRINTME("x3", x3);
+    PRINTME("x4", x4);
+
+    // Schur-type composition
+    SparseMatrix<double> B (sparsity_pattern);
+    SparseMatrix<double> C (sparsity_pattern);
+    SparseMatrix<double> D (sparsity_pattern);
+
+    for (unsigned int i=0; i < rc; ++i)
+      {
+        B.diag_element(i) = 4.0;
+        C.diag_element(i) = 4.0; // symmetric
+        D.diag_element(i) = 3.0;
+      }
+
+    const auto lo_B = linear_operator(B);
+    const auto lo_C = linear_operator(C);
+    const auto lo_D = linear_operator(D);
+    const auto lo_B_t = transpose_operator(lo_B);
+    const auto lo_C_t = transpose_operator(lo_C);
+    const auto lo_D_t = transpose_operator(lo_D);
+
+    deallog << "Single packaged operation" << std::endl;
+    {
+      const auto S = lo_D - lo_C*lo_A_inv*lo_B;
+      const auto S_t = lo_D_t - lo_B_t *lo_A_inv_t *lo_C_t;
```

Comment:
@tamiko The result of operating `S` and `transpose(S)` on a vector should be the same as `B==C`. It would appear that this line, with the packaged operation involving an transpose inverse matrix leads to an incorrect result. When one deconstructs the operation and does each of the underlying steps manually, one gets the right result.